### PR TITLE
(feat) Make the label in FormField optional to support Markdown content

### PR DESCRIPTION
**Pull request by Twiineenock**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

### Problem Description
This pull request is part of ongoing work in [PR #355](https://github.com/openmrs/openmrs-esm-form-builder/pull/355).

In the interactive form builder, users should have the ability to create questions with markdown-rendered content. The markdown feature will allow question content to be formatted for better readability and accessibility.

### Key Changes

1. **Upgrade `react-markdown` Version**:
   - Upgraded `react-markdown` from version `^7.1.2` to `^9.0.1` to support an expanded range of markdown formatting options.
   - This update enables support for additional plugins and formatting elements, allowing for a richer editing experience.

2. **Extended Markdown Capabilities**:
   - Extended the `ReactMarkdown` component to support more plugins and `allowedElements` for enhanced markdown rendering.
   - Updated the `FormField` interface to handle cases where the `label` is absent, but a `value` is present. This change provides more flexibility for displaying markdown content without requiring a label.

### Technical Details

- **Dependencies**:
  ```json
  "react-markdown": "^9.0.1"


## Screenshots
<!-- Required if you are making UI changes. -->

## Related Issue
<!-- Paste the link to the Jira ticket here if one exists. -->
<!-- https://issues.openmrs.org/browse/O3- -->
[https://openmrs.atlassian.net/browse/O3-3946](https://openmrs.atlassian.net/browse/O3-3946)

## Other
<!-- Anything not covered above -->

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,9 +34,11 @@
     "lodash-es": "^4.17.21",
     "react-error-boundary": "^4.0.13",
     "react-hook-form": "^7.52.0",
-    "react-markdown": "^7.1.2",
+    "react-markdown": "^9.0.1",
     "react-waypoint": "^10.3.0",
-    "react-webcam": "^7.2.0"
+    "react-webcam": "^7.2.0",
+    "rehype-raw": "^7.0.0",
+    "remark-gfm": "^4.0.0"
```
Comment (NethmiRodrigo) [inline package.json:41]: Do we need these two libraries?

File: src/types/schema.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export interface FormSection {
 }
 
 export interface FormField {
-  label: string;
+  label?: string;
```
Comment (NethmiRodrigo) [inline src/types/schema.ts:60]: @samuelmale this was changed to be optional because markdown type inputs don't have a label. Does this seem okay?

**Comment by NethmiRodrigo** [comment]
@Twiineenock can you resolve the conflicts and update your branch?

File: src/types/schema.ts (hunk 2 above)
Comment (samuelmale) [reply src/types/schema.ts:60]: That makes sense to me.

File: src/components/inputs/markdown/markdown-wrapper.component.tsx [hunk 3]
```diff
@@ -6,7 +6,17 @@ const MarkdownWrapper: React.FC<{ markdown: string | string[] }> = ({ markdown }
...
+        'h1',
+        'h2',
+        'h3',
+        'h4',
+        'h5',
+        'h6',
+        'p',
+        'strong',
+        'em',
+      ]}
```
Comment (denniskigen) [inline src/components/inputs/markdown/markdown-wrapper.component.tsx:19]: @Twiineenock AFAICT, it doesn't seem as though the allowedElements have changed per this this diff. Does that invalidate the PR title?  Is the main change here that we're just bumping `react-markdown`?

**Review by denniskigen — COMMENTED** [review]
Thanks for working on this. I've left a comment seeking some clarification about the changes here.

File: src/components/inputs/markdown/markdown-wrapper.component.tsx (hunk 3 above)
Comment (Twiineenock) [reply src/components/inputs/markdown/markdown-wrapper.component.tsx:19]: And yes the PR title has to change! Thanks!

**Review by NethmiRodrigo — APPROVED** [review]

**Comment by NethmiRodrigo** [comment]
LGTM! Thanks @Twiineenock!

**Pull request merged**